tpp-app/src/services/CalendarService.js

```diff
@@ -0,0 +1,120 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { KEYS, TRACK_SYMPTOMS } from './utils/constants.js'
+import { Symptoms } from './utils/models.js';
+
+/**
+ * Makes API request to AsyncStorage to access symptoms for the entire year
+ * @param year: number
+ * @returns: object
+ */
+export const GETYearData = async (year) => {
+    // https://github.com/uoftblueprint/the-period-purse/pull/95#discussion_r813506726
+    // Essentially retrieve the calendar for a given year and convert each day's data into a Symptom object for better front-end handling
+
+    // Used for testing
+    // await AsyncStorage.setItem('2022', JSON.stringify(
+    //     [
+    //         // 0th index of the 2022 array is 1st month i.e. January has size 31
+    //         [
+    //             {
+    //                 'Flow': 'LIGHT',
+    //                 'Mood': 'HAPPY',
+    //                 'Sleep': '7.5', 
+    //                 'Cramps': 'MEDIUM', 
+    //             }
+    //         ]
+    //     ])
+    // )    
+
+    const yearData = JSON.parse(await AsyncStorage.getItem(year.toString()));
+
+    // yearData could be null
+    if (yearData) {
+        yearData.map((month) => {
+            month.map((day) => {
+                return day ? new Symptoms(day) : new Symptoms();
+            })
+        })
+        return yearData;
+    }
+
+    return null;
+}
+
+/**
+ * Saves user's current selected filter, selected month, and selected year, to preserve the thing they are looking at.
+ * Meant to be used when user exits calendar page in any way.
+ * @param selectedView: string
+ * @param selectedMonth: number
+ * @param selectedYear: number
+ */
+
+export const POSTMostRecentCalendarState = async (selectedView, selectedMonth, selectedYear) => new Promise( async (resolve, reject) => {
+
+    // Check if it's a valid view or month
+    if (selectedMonth > 12 || selectedMonth < 1) {
+        reject("No month to record")
+    }
+
+    var exists = Object.keys(TRACK_SYMPTOMS).some(function(k) {
+        return TRACK_SYMPTOMS[k] === selectedView;
```

Comment:
1. Small note but try and avoid using `var` because it usually implies a global variable. It's actually fine here because it's functionally scoped, but overall in JavaScript it's good practice to stick with `let` or `const` unless you want it to be globally-scoped, in which case we use `var`. Ref: [MDN doc for var](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/var)

2. Since we're sticking with arrow notation I recommend the following, where `tracking` is equivalent to `TRACK_SYMPTOMS[k]`.

```suggestion
    const exists = Object.keys(TRACK_SYMPTOMS).some((tracking) => tracking === selectedView);
```